x-pack/plugins/security_solution/docs/testing/test_plans/detection_response/alerts/user_assignment.md

```diff
@@ -0,0 +1,186 @@
+```
+
+#### **Scenario: Update assignees**
+
+**Automation**: 3 e2e test + 1 unit test + 1 integration test.
+
+```Gherkin
+Given an alert with assignees
+When user removes some of (or all) current assignees and adds new assignees
+Then assignees field should be updated
+And removed assignees should be absent
+And newly added assignees should be present
+```
+
+#### **Scenario: Remove all assignees**
+
+**Automation**: 2 e2e test + 1 unit test.
+
+```Gherkin
+Given an alert with assignees
+When user triggers "Remove all assignees" action
+Then assignees field should be updated
+And assignees field should be empty
+```
+
+### Updating assignees (bulk actions)
+
+#### **Scenario: Add new assignees**
+
+**Automation**: 1 e2e test + 1 unit test + 1 integration test.
+
+```Gherkin
+Given multiple alerts
+When user adds new assignees
+Then assignees fields of all involved alerts should be updated
+And newly added assignees should be present
+```
+
+#### **Scenario: Update assignees**
+
+**Automation**: 1 e2e test + 1 unit test + 1 integration test.
+
+```Gherkin
+Given multiple alerts with assignees
+When user removes some of (or all) current assignees and adds new assignees
+Then assignees fields of all involved alerts should be updated
+And removed assignees should be absent
+And newly added assignees should be present
+```
+
+#### **Scenario: Remove all assignees**
+
+**Automation**: 1 e2e test + 1 unit test.
+
+```Gherkin
+Given multiple alerts with assignees
+When user triggers "Remove all assignees" action
+Then assignees fields of all involved alerts should be updated
+And assignees fields should be empty
+```
+
+### Alerts filtering
+
+#### **Scenario: By one assignee**
+
+**Automation**: 1 e2e test + 1 unit test.
+
+```Gherkin
+Given multiple alerts with and without assignees
+When user filters by one of the assignees
+Then only alerts with selected assignee in assignees field are displayed
+```
+
+#### **Scenario: By multiple assignees**
+
+**Automation**: 1 e2e test + 1 unit test.
+
+```Gherkin
+Given multiple alerts with and without assignees
+When user filters by multiple assignees
+Then all alerts with either of selected assignees in assignees fields are displayed
+```
+
+#### **Scenario: "No assignees" option**
+
+**Automation**: 1 e2e test + 1 unit test.
+
+```Gherkin
+Given filter by assignees UI is available
+Then there should be an option to filter alerts to see those which are not assigned to anyone
+```
+
+#### **Scenario: By "No assignees"**
+
+**Automation**: 1 e2e test + 1 unit test.
+
+```Gherkin
+Given multiple alerts with and without assignees
+When user filters by "No assignees" option
+Then all alerts with empty assignees fields are displayed
+```
+
+### Authorization / RBAC
+
+#### **Scenario: Viewer role**
+
+**Automation**: 2 e2e test + 1 unit test + 1 integration test.
+
+```Gherkin
+Given user has VIEWER role
+Then there should not be a way to update assignees field for an alert
+```
```

Comment:
I would add use case for all the prebuilt serverless roles too.

Reply:
Is [this](https://www.elastic.co/guide/en/cloud/current/ec-user-privileges.html) and [this](https://www.elastic.co/guide/en/cloud/current/ec-stack-user-org-member.html) are right links with Serverless roles descriptions?

Reply:
Or [this one](https://github.com/elastic/project-controller/blob/main/internal/project/security/config/roles.yml) covers roles better? Found a nice grid with roles in one of yours [PRs](https://github.com/elastic/kibana/pull/170380) :-)

Reply:
Looks like all the prebuilt roles (`t1_analyst`, `t2_analyst`, `t3_analyst`, `threat_intelligence_analyst`, `rule_author`, `soc_manager`, `detections_admin`, `platform_engineer`, `endpoint_operations_analyst`, `endpoint_policy_manager`) except the `viewer` are able to modify alerts index. Am I right or did I miss something?

Reply:
done

Reply:
Sorry I missed this Zhenia. Looks like you already figured it out! Yes, all except viewer can modify alerts.